src/GitVersionCore/Core/Abstractions/IDisposableServiceProvider.cs

```diff
@@ -0,0 +1,7 @@
+using System;
+
+namespace GitVersion.Core.Abstractions
+{
+    public interface IDisposableServiceProvider : IServiceProvider, IDisposable
```

Comment:
I think ServiceProvider class that implements IServiceProvider is already implementing IDisposable. So I would not add an additional interface for that

Reply:
Why do you not confirm yourself? :) You can see [here in the documentation](https://docs.microsoft.com/en-US/dotnet/api/system.iserviceprovider?view=netcore-3.1) of Microsoft that IServiceProvider is not implementing it. When IServiceProvider is not implementing IDisposable, what's your approach? Return ServiceProvider directly?

Reply:
`IServiceProvider` does not inherit `IDisposable`, correct. But [the `ServiceProvider` class implements `IDisposable`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceprovider):

```c#
public sealed class ServiceProvider : IAsyncDisposable, IDisposable, IServiceProvider
```

Reply:
Ok I might be right, when I say, that it is okay for you to return only IDisposable, right?

Reply:
Right. So returning ServiceProvider directly should be fine.

Reply:
If that means the `IDisposableServiceProvider` will be deleted, I think we agree, @teroneko! 😃